### PR TITLE
Fixed type resolution for anchormaps

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Fixed anchormap
# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [ x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/ast/statements/call.py [hunk 1]
```diff
@@ -83,8 +83,8 @@ def __init__(
 
     def normalize(self, *, lhs_attribute: Optional[AttributeAssignmentLHS] = None) -> None:
         ReferenceStatement.normalize(self)
-        self.anchors = [TypeReferenceAnchor(self.namespace, self.name)]
         func = self.namespace.get_type(self.name)
+        self.anchors = [TypeReferenceAnchor(self.namespace, self.name)]
```
Comment (wouterdb) [inline src/inmanta/ast/statements/call.py:87]: Typeanchor acted up here,so I left is as it is

Comment (FloLey) [reply src/inmanta/ast/statements/call.py:87]: Do you have an idea why?

File: tests/test_compiler_entrypoints.py [hunk 2]
```diff
@@ -346,3 +346,37 @@ def test_constructor_with_inferred_namespace(snippetcompiler):
     range_target = Range(target_path, 1, 8, 1, 12)
 
     assert (range_source, range_target) in anchormap
+
+
+def test_constructor_renamed_namespace(snippetcompiler):
+    """
+    Test that the anchor for a constructor with `import a as b` work
```
Comment (FloLey) [inline tests/test_compiler_entrypoints.py:353]:
```suggestion
    Test that the anchor for a constructor with `import a as b` works
```

**Review by FloLey — APPROVED** [review]

File: src/inmanta/ast/statements/call.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/ast/statements/call.py:87]: yes, this is not quite a type, it might work is I move it a few lines down, but I didn't want to take risks

**Pull request closed without merging**